extensions/guacamole-auth-nextcloud/src/main/java/org/apache/guacamole/auth/nextcloud/NextcloudJwtAuthenticationProvider.java

```diff
@@ -0,0 +1,189 @@
+        String ipaddr = request.getRemoteAddr();
+
+        boolean localAddr = this.validIpAddress(ipaddr);
+        if (localAddr) {
+            logger.info("Request from local address {}", ipaddr);
+            return null;
+        }
+
+        if (token == null) {
+            throw new GuacamoleException("Missing token.");
+        }
+
+        try {
+            boolean valid = this.isValidJWT(token);
+            if (!valid) {
+                throw new GuacamoleException("Token expired.");
+            }
+            logger.info("Token valid.");
+        } catch (final GuacamoleException ex) {
+            logger.error("Token validation failed.", ex);
+            throw new GuacamoleException(ex.getMessage());
+        }
+        return null;
+
+    }
+
+    /**
+     * Validates the provided JSON Web Token (JWT).
+     *
+     * This method decodes the public key from a base64 encoded string, verifies the JWT using
+     * the ECDSA256 algorithm, and checks the token's validity period and user permissions.
+     *
+     * @param token the JWT token to validate.
+     * @return {@code true} if the token is valid and the user is allowed, {@code false} otherwise.
+     * @throws GuacamoleException if the user is not allowed or the token is expired.
+     * @throws JWTVerificationException if the token verification fails.
+     * @throws NoSuchAlgorithmException if the algorithm for key generation is not available.
+     * @throws InvalidKeySpecException if the key specification is invalid.
+     */
+    private boolean isValidJWT(final String token) throws GuacamoleException {
+        try {
+            byte[] keyBytes = Base64.getDecoder().decode(confService.getPublicKey());
+            KeyFactory keyFactory = KeyFactory.getInstance("EC");
+            X509EncodedKeySpec keySpec = new X509EncodedKeySpec(keyBytes);
+            ECPublicKey publicKey = (ECPublicKey) keyFactory.generatePublic(keySpec);
+
+            JWTVerifier verifier = JWT.require(Algorithm.ECDSA256(publicKey)).build();
+            DecodedJWT decodedJWT = verifier.verify(token);
+
+            Date currentDate = new Date();
+            Date maxValidDate = new Date(currentDate.getTime() - (MINUTES_TOKEN_VALID * 60 * 1000));
+            boolean isUserAllowed = this.isUserAllowed(decodedJWT.getPayload());
+            if (!isUserAllowed) {
+                throw new GuacamoleException("User not allowed.");
+            }
+
+            boolean isValidToken = decodedJWT.getExpiresAt().after(maxValidDate);
+            if (!isValidToken) {
+                throw new GuacamoleException("User not allowed.");
+            }
+
+            return true;
+        } catch (final JWTVerificationException | NoSuchAlgorithmException | InvalidKeySpecException ex) {
+            logger.error("Token validation failed.", ex);
+            return false;
+        }
+    }
+
+    private boolean validIpAddress(final String ipAddress) throws GuacamoleException {
```

Comment:
As @jmuehlner indicated, please document this method - both declaration and throughout the code.

Reply:
Please put empty lines between `@param`, `@return`, and `@throws`.